### DISCUSSION TITLE
Add branch param for pr-publish to set ref

### DISCUSSION THREAD

**Discussion by WingLim** [Homebrew/discussions#1492] [Tap maintenance and Homebrew development]
In here: https://github.com/Homebrew/brew/blob/2e9d4ffa157a11febc928eb4e87ea5a46bb9293a/Library/Homebrew/dev-cmd/pr-publish.rb#L39

The ref is hardcode to `master`, it works for the `homebrew-core`.

But if we create a tap for ouselves, we may use `main` or other branch name as default.

If so, we can not use `brew pr-publish` for tap.

It will be good to add a `branch` param for `pr-publish`.

Set the default to `master`, keeping `homebrew-core` works.

**Comment by carlocab** [comment]
We'll review a pull request for this.

**Reply by WingLim** [reply]
Ok, I create a pr in https://github.com/Homebrew/brew/pull/11410.
So, take a look.
